ts/screens/wallet/AddCardScreen.tsx

```diff
@@ -89,9 +90,29 @@ const INITIAL_STATE: State = {
   pan: none,
   expirationDate: none,
   securityCode: none,
-  holder: none
+  holder: none,
+  isValidHolder: undefined,
+  isValidPan: undefined,
+  isValidExpirationDate: undefined,
+  isValidSecurityCode: undefined
 };
 
+function isExpirationDateValid(expirationDate: string): boolean {
+  const [expirationMonth, expirationYear] = expirationDate.split("/");
+  return (
+    CreditCardExpirationMonth.is(expirationMonth) &&
+    CreditCardExpirationYear.is(expirationYear)
+  );
+}
+
+function isSecurityCodeValid(securityCode: string): boolean {
+  return !CreditCardCVC.is(securityCode);
+}
+
+function isPanValid(pan: string): boolean {
```

Comment:
I can't figure out how to avoid type error when using this function. It happens if I use this function to check the pan validity also in the getCardFromState function (line 129). The error occurs on the declaration of the card pan (line 147):
"Type 'string' is not assignable to type 'string & IPatternStringTag<string>'"

 it's like it doesn't verify if the type check is granted by the isPanValid function.

@cloudify Any suggestion on how to accomplish it?

Reply:
Yes, the error means that the `pan` is not a simple string, but it's instead a _tagged_ string (a string with a special meaning), in this case is a string that conforms to a certain pattern (the PAN). The `CreditCard` type (declared in `types/pagopa.ts` is indeed composed of tagged string types declared [here](https://github.com/teamdigitale/io-app/blob/master/ts/utils/input.ts) - you should use `CreditCardPan` instead of `string` (same for expiration date and all other card fields)

Reply:
thank you, it was simpler than expected. Using "as CreditCardPanType" on credit card initialization solves the problem. I don't use the CreditCardPan type for the pan state being this state used to store the user input.